src/odemis/util/concurrent.py

```
@@ -0,0 +1,430 @@
+                            fn, args, kwargs = work
+                    except queue.Empty:
+                        pass  # No more work in the queue => everything is fine
+
+                fn(*args, **kwargs)
+        except Exception:
+            logging.exception("Error in BackgroundWorker with function %s", fn)
+        finally:
+            logging.debug("BackgroundWorker thread finished")
+
+
+def executeAsyncTask(future, fn: Callable, args: tuple = (), kwargs: dict = None) -> threading.Thread:
```

Ruff (0.15.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)